gdsfactory/routing/route_ports_to_side.py

```diff
@@ -86,23 +87,34 @@ def route_ports_to_side(
     if not ports:
         return [], []
 
-    if side == "north" or side == "south":
-        xy_ns = y if y is not None else side
+    if side in {"north", "south"}:
```

Comment:
Not sure if all of this is needed, ik the types are very weird here with literals but what we had did work, it avoids all of this casting etc.

Reply:
yes, im surprised passing kwargs made what we had no longer work,

i can merge it and we can make some improvements later if needed